cg/services/pacbio_services/pacbio_metrics_service/metrics_parser.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+from typing import Any, Callable
+
+from cg.constants.constants import FileFormat
+from cg.constants.pacbio import PacBioDirsAndFiles
+from cg.io.controller import ReadFile
+from cg.services.pacbio_services.pacbio_metrics_service.models import HiFiMetrics
+from cg.utils.files import get_file_in_directory
+
+
+class MetricsParser:
+    """Class for parsing PacBio sequencing metrics."""
+
+    def __init__(self, smrt_cell_path: Path) -> None:
+        self.smrt_cell_path: Path = smrt_cell_path
+        self.report_path = Path(smrt_cell_path, "statistics")
+        # For HiFi metrics
+        self.css_report_path: Path = get_file_in_directory(
+            directory=self.report_path, file_name=PacBioDirsAndFiles.CCS_REPORT
+        )
+        # For control metrics
+        self.control_report_path: Path = get_file_in_directory(
+            directory=self.report_path, file_name=PacBioDirsAndFiles.CONTROL_REPORT
+        )
+        # For productivity metrics
+        self.loading_report_path: Path = get_file_in_directory(
+            directory=self.report_path, file_name=PacBioDirsAndFiles.LOADING_REPORT
+        )
+        # For polymerase metrics
+        self.raw_data_report_path: Path = get_file_in_directory(
```

Comment:
Not sure if we've discussed this in infrastructure way back but I think type hinting as `Path` is enough. No need to name the varialbe `_path`

Reply:
when referring to paths we use path, otherwise it can be interpreted as model

Reply:
Would it not be nice to use `_file` to distinguish from `_dir`. Then. it is very clear and just as readable.

Reply:
changed to `_dir` and `_file`